internal/upload/azure/azure.go

```diff
@@ -1,158 +1,149 @@
-
-	// get storage account blob service URL endpoint.
-	URL, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", credentials.StorageAccount, metadata.ContainerName))
+	return &Client{
+		authorizer: authorizer,
+	}, nil
+}
 
-	// Create a ContainerURL object that wraps the container URL and a request
-	// pipeline to make requests.
-	containerURL := azblob.NewContainerURL(*URL, p)
+// Tag is a name-value pair representing the tag structure in Azure
+type Tag struct {
+	Name  string
+	Value string
+}
 
-	// Create the container, use a never-expiring context
-	ctx := context.Background()
+// GetResourceNameByTag returns a name of an Azure resource tagged with the
+// given `tag`. Note that if multiple resources with the same tag exists
+// in the specified resource group, only one name is returned. It's undefined
+// which one it is.
+func (ac Client) GetResourceNameByTag(ctx context.Context, subscriptionID, resourceGroup string, tag Tag) (string, error) {
+	c := resources.NewClient(subscriptionID)
+	c.Authorizer = ac.authorizer
 
-	// Open the image file for reading
-	imageFile, err := os.Open(fileName)
+	filter := fmt.Sprintf("tagName eq '%s' and tagValue eq '%s'", tag.Name, tag.Value)
+	result, err := c.ListByResourceGroup(ctx, resourceGroup, filter, "", nil)
 	if err != nil {
-		return fmt.Errorf("cannot open the image: %v", err)
+		return "", fmt.Errorf("listing resources failed: %v", err)
 	}
-	defer imageFile.Close()
 
-	// Stat image to get the file size
-	stat, err := imageFile.Stat()
-	if err != nil {
-		return fmt.Errorf("cannot stat the image: %v", err)
+	if len(result.Values()) < 1 {
+		return "", nil
 	}
 
-	// Hash the imageFile
-	imageFileHash := md5.New()
-	if _, err := io.Copy(imageFileHash, imageFile); err != nil {
-		return fmt.Errorf("cannot create md5 of the image: %v", err)
+	return *result.Values()[0].Name, nil
+}
+
+// CreateStorageAccount creates a storage account in the specified resource
+// group. The location parameter can be used to specify its location. The tag
+// can be used to specify a tag attached to the account.
+func (ac Client) CreateStorageAccount(ctx context.Context, subscriptionID, resourceGroup, name, location string, tag Tag) error {
+	c := storage.NewAccountsClient(subscriptionID)
+	c.Authorizer = ac.authorizer
+
+	result, err := c.Create(ctx, resourceGroup, name, storage.AccountCreateParameters{
+		Sku: &storage.Sku{
+			Name: storage.StandardRAGRS, // TODO: investigate the default value
+			Tier: storage.Standard,
+		},
+		Location: &location,
+		Tags: map[string]*string{
+			tag.Name: &tag.Value,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("sending the create storage account request failed: %v", err)
 	}
-	// Move the cursor back to the start of the imageFile
-	if _, err := imageFile.Seek(0, 0); err != nil {
-		return fmt.Errorf("cannot seek the image: %v", err)
+
+	err = result.WaitForCompletionRef(context.Background(), c.Client)
```

Comment:
Shouldn't the method's `ctx` argument be used here as well?

Reply:
Fixed in the whole `azure.go` file.